src/compiler/output-targets/dist-custom-elements/test/dist-custom-elements.spec.ts

```diff
@@ -0,0 +1,25 @@
+import { addCustomElementInputs } from '../index';
+
+describe('dist-custom-elements', () => {
+  it('should add custom element inputs', () => {
+    const buildCtx: any = {
+      components: [
+        {
+          isPlain: true,
+          sourceFilePath: './foo/bar.tsx',
+          tagName: 'my-tag',
+        },
+      ],
+    };
```

Comment:
Note: This is included in the patch file in my first comment on this PR


To build out a mock `BuildContext`, we have `mockBuildCtx` ([GH Link](https://github.com/ionic-team/stencil/blob/6faf746990330da4369e0d73725b0fc2becebb33/src/testing/mocks.ts#L199)). This builds a build context with a `ValidatedConfig` and `CompilerCtx` for us, and defaults to known good versions if neither are provided.

At the moment, we can't override values like `components` on it, but we can directly assign that property on the `buildCtx` object, which is the only property used by `addCustomElementIntputs`.

To build out an array of one of `ComponentCompilerMeta`s, we can use `stubComponentCompilerMeta` ([GH Link](https://github.com/ionic-team/stencil/blob/5976c9b6a6e7b49d470390021b9c31e4d3cbbf4b/src/compiler/types/tests/ComponentCompilerMeta.stub.ts#L10)). Unlike `mockBuildCtx`, we can override just about anything on this object.

All in all, I think we could accomplish the same thing with a bit more type safety with:
```ts
    const cmpMeta = stubComponentCompilerMeta({ isPlain: true, sourceFilePath: './foo/bar.tsx', tagName: 'my-tag' });
    const buildCtx = mockBuildCtx();
    buildCtx.components = [cmpMeta];
```